src/language-js/printer-estree.js

```diff
@@ -5713,10 +5710,13 @@ function printBinaryishExpressions(
   isInsideParenthesis
 ) {
   let parts = [];
-  const node = path.getValue();
+  const node = convertToBinaryishNode(path.getValue());
 
   // We treat BinaryExpression and LogicalExpression nodes the same.
   if (isBinaryish(node)) {
+    const leftNodeName = node.type === "TSAsExpression" ? "expression" : "left";
+    const rightNodeName =
+      node.type === "TSAsExpression" ? "typeAnnotation" : "right";
```

Comment:
Do we really need `convertToBinaryishNode` if we have such variables (`leftNodeName` and `rightNodeName`)?

Reply:
I thought it would lower the cost of maintenance, but given the problems you've shown above, I think it's unnecessary.